packages/fx-core/src/core/middleware/projectMigrator.ts

```diff
@@ -444,6 +447,12 @@ async function migrateMultiEnv(projectPath: string): Promise<void> {
   await localSettingsProvider.save(
     localSettingsProvider.init(hasFrontend, hasBackend, hasBot, migrateFromV1)
   );
+  await addPathToGitignore(projectPath, localSettingsProvider.localSettingsFilePath);
```

Comment:
Move this logic to `postMigration` method or recover it in `cleanup` method. 
Cause if upgrade failed, user shouldn't see it in .gitignore file.

Reply:
Yeah, it makes sense to me.